### ISSUE TITLE
Revert notebook to saved - confirmation pop-up

### ISSUE THREAD

**Issue by radugrosu** [jupyterlab/jupyterlab#4941]
_Revert notebook to saved_  is dangerously close to _Rename notebook_  in the _File_ menu, and executes its action without confirmation. This adds to the number of ways in which one can lose work while using Jupyter, in this case according to the workflow: Idea -> _Untitled.ipynb_ -> Implementation -> _Revert notebook to saved_ (mistakenly, instead of _Rename notebook_).
A quick fix would be to ask for confirmation on any action with such destructive potential.

**Comment by blink1073** [comment]
Good catch, thanks!  Most other editors will revert without prompting, provided the changes are available in your change history, but that is not so easy with a notebook document.  Here is the dialog from the classic notebook:

<img width="702" alt="image" src="https://user-images.githubusercontent.com/2096628/43069638-3cee3620-8e33-11e8-9896-95611e143e2b.png">

And [here](https://github.com/jupyterlab/jupyterlab/blob/8fe2da16960cc05d271cc70a9e8f5aa2b4d28926/packages/docmanager-extension/src/index.ts#L412) is the part of the code that needs updating to use `showDialog`.

**Comment by kcramer** [comment]
I was about to submit a PR for this issue but right before submitting it, I realized these comments could apply both to "Reload Notebook from Disk" and "Revert Notebook to Checkpoint".  I coded a confirmation only for the revert to checkpoint.

The concern over losing data could apply to either one.  The menu items have changed recently so it was not clear on the intent.  It wouldn't be too difficult to add a confirmation for the Reload too.
